packages/sanity/src/core/store/_legacy/document/document-pair/utils/fetchFeatureToggle.ts

```diff
@@ -0,0 +1,35 @@
+import {type SanityClient} from '@sanity/client'
+import {map, type Observable, of, ReplaySubject, timer} from 'rxjs'
+import {catchError, share} from 'rxjs/operators'
+
+interface ActionsFeatureToggle {
+  actions: boolean
+}
+
+//in the "real" code, this would be observable.request, to a URI
+export const fetchFeatureToggle = (defaultClient: SanityClient): Observable<boolean> => {
+  const client = defaultClient.withConfig({apiVersion: 'X'})
+  const {dataset} = defaultClient.config()
+
+  return client.observable
+    .request({
+      uri: `/data/actions/${dataset}`,
+      withCredentials: true,
+    })
+    .pipe(
+      map((res: ActionsFeatureToggle) => res.actions),
+      catchError(() =>
```

Comment:
Do you think there's any value in adding a timeout as a safeguard here? My thinking is that slow feature flag resolution should never negatively impact user experience.

Reply:
(I know this is a bit of a moot point at the moment given that feature flags are fairly static and powered by the same backend as other Studio features—but they will probably become more complicated in the future).

Reply:
That's a great point. I don't think it hurts to add a timeout here. Added a 2s timeout in 0de508717a - if the request takes longer than 2s we'll treat it as feature "off"

Reply:
Wonderful!